connectors/protocol/connectors.py

```diff
@@ -662,22 +673,38 @@ async def error(self, error):
     async def sync_done(self, job, cursor=None):
         job_status = JobStatus.ERROR if job is None else job.status
         job_error = JOB_NOT_FOUND_ERROR if job is None else job.error
+        job_type = job.job_type if job is not None else None
```

Comment:
This is an edge case, which I don't know how to handle

E.g. An access control sync runs, and somehow the sync job is deleted. `sync_done` is called with the `None` job. In this case, `last_access_control_sync_status` will not be updated, and it will block other access control sync to be executed.

We could choose to reset both `last_access_control_sync_status` and `last_sync_status`. The worst case is that, there is another full/incremental sync running, and this can possibly execute another full/incremental sync. But I think it's still better the the previous mention case.

Reply:
I think this is a good trade-off, I would also prefer to rerun syncs in the worst case vs. blocking access control syncs 👍